From ea4fc3e66cfb471d25b31e57e82f702d7ff22f67 Mon Sep 17 00:00:00 2001
From: Roy Pledge <Roy.Pledge@freescale.com>
Date: Tue, 16 Jul 2013 10:32:11 -0400
Subject: [PATCH 277/987] Fix regression that caused BMan interrupt lookup for
 USDPAA to return incorrect information

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

Signed-off-by: Roy Pledge <Roy.Pledge@freescale.com>
Change-Id: I72e62fb6d37f63d13bd1f93029f1a4af24924ba9
Reviewed-on: http://git.am.freescale.net:8181/3336
Reviewed-by: Wang Haiying-R54964 <Haiying.Wang@freescale.com>
Reviewed-by: Thorpe Geoff-R01361 <Geoff.Thorpe@freescale.com>
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Fleming Andrew-AFLEMING <AFLEMING@freescale.com>
---
 drivers/staging/fsl_qbman/fsl_usdpaa.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/fsl_qbman/fsl_usdpaa.c b/drivers/staging/fsl_qbman/fsl_usdpaa.c
index 7df18e1..c19f3f3 100644
--- a/drivers/staging/fsl_qbman/fsl_usdpaa.c
+++ b/drivers/staging/fsl_qbman/fsl_usdpaa.c
@@ -1294,11 +1294,14 @@ int usdpaa_get_portal_config(struct file *filp, void *cinh,
 	list_for_each_entry(portal, &context->portals, list) {
 		if (portal->user.type == ptype &&
 		    portal->user.addr.cinh == cinh) {
+			if (ptype == usdpaa_portal_qman) {
+				*irq = portal->qportal->public_cfg.irq;
+				*iir_reg = portal->qportal->addr_virt[1] + QM_REG_IIR;
+			} else {
+				*irq = portal->bportal->public_cfg.irq;
+				*iir_reg = portal->bportal->addr_virt[1] + BM_REG_IIR;
+			}
 			spin_unlock(&context->lock);
-			*irq = portal->qportal->public_cfg.irq;
-			*iir_reg = portal->qportal->addr_virt[1] +
-				((ptype == usdpaa_portal_qman) ? QM_REG_IIR :
-								 BM_REG_IIR);
 			return 0;
 		}
 	}
-- 
1.9.1

